packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -250,6 +256,12 @@ class DatePicker extends React.Component {
     delete customProps.onCalendarButtonClick;
     delete customProps.onInputFocus;
 
+    const attributes = Object.assign({}, customProps);
```

Comment:
`customProps` is already be a local object. You should be able to just conditionally add the aria required attribute directly to customProps. ( If we want to keep the aria attribute pending the other discussion )

```
if (required) {
  customProps['aria-required'] = 'true';
}
```

Reply:
Your edit comment is correct, assignment was removed in this commit: 14324e7